src/Android/Xamarin.Android/Samples/Layers/RasterRenderingRule/RasterRenderingRule.cs

```diff
@@ -0,0 +1,173 @@
+// Copyright 2017 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at: http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an 
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific 
+// language governing permissions and limitations under the License.
+
+using Android.App;
+using Android.OS;
+using Android.Views;
+using Android.Widget;
+using Esri.ArcGISRuntime.ArcGISServices;
+using Esri.ArcGISRuntime.Geometry;
+using Esri.ArcGISRuntime.Mapping;
+using Esri.ArcGISRuntime.Rasters;
+using Esri.ArcGISRuntime.UI.Controls;
+using System;
+using System.Collections.Generic;
+
+namespace ArcGISRuntimeXamarin.Samples.RasterRenderingRule
+{
+    [Activity]
+    public class RasterRenderingRule : Activity
+    {
+        // Create and hold reference to the used MapView
+        private MapView _myMapView = new MapView();
+
+        // Create a global scope empty read-only list for the various rendering rules of the image service raster
+        private IReadOnlyList<RenderingRuleInfo> _myReadOnlyListRenderRuleInfos;
+
+        // Create a global scope Uri for the image server
+        private Uri _myUri;
+
+        // Create a global list to store the names of the rendering rule info for the image service raster
+        private List<string> _names = new List<string>();
+
+        // Create a global scope button for the user to change the rendering rule for the image service raster
+        private Button _renderingRulesButton;
+
+        protected override void OnCreate(Bundle bundle)
+        {
+            base.OnCreate(bundle);
+
+            Title = "Raster rendering rule";
+
+            // Create the UI, setup the control references and execute initialization 
+            CreateLayout();
+            Initialize();
+        }
+
+        private async void Initialize()
+        {
+            // Assign a new map to the MapView
+            _myMapView.Map = new Map();
+
+            // Set the basemap to Streets
+            _myMapView.Map.Basemap = Basemap.CreateStreets();
+
+            // Set the Uri for the image server
+            _myUri = new Uri("https://sampleserver6.arcgisonline.com/arcgis/rest/services/CharlotteLAS/ImageServer");
```

Comment:
The pattern I've seen in other samples is to move the Uri to a class-level variable. I don't know if that needs to change in this case but maybe there should be a style guide for such things?

Reply:
After re-reading I see that this is a class-scope variable. I would prefer if this was initialized where it is declared rather than in a separate variable. If the user sees this referenced in another function, they will need to search for the code that assigns the value rather than just looking at the declaration.